src/compile.rs

```diff
@@ -2084,49 +2118,74 @@ pub fn call_derived_associated_calls<'a>(
         if !t.is_domain(types) {
             continue;
         }
-        for a in &t.annotations {
-            if let AnnotationInfo::Associate(associations) = a {
-                for f in funcs.values() {
-                    if f.is_derived && f.is_associated_call {
-                        let resource_name = match f.class {
-                            FunctionClass::Type(n) => n.name.clone(),
-                            _ => {
-                                // Can't derive from Global or API
-                                continue;
-                            }
-                        };
-                        if associations.resources.iter().any(|r| {
-                            [t.name.as_ref(), r.as_ref()].join(".") == resource_name.as_ref()
-                        }) {
-                            let call = make_associated_call(resource_name, f);
-                            let args = vec![FunctionArgument::new_this_argument(t)];
-                            // TODO: Should there be a parent_context here?  I think this is
-                            // effectively a "fake" context since we're not really parsing the tree
-                            let mut local_context =
-                                BlockContext::new(BlockType::Domain, Some(t), None);
-                            local_context.insert_function_args(&args);
-
-                            let validated_calls = match ValidatedCall::new(
-                                &call,
-                                funcs,
-                                types,
-                                class_perms,
-                                &local_context,
-                                f.declaration_file,
-                            ) {
-                                Ok(c) => c.inner(&mut warnings),
-                                Err(e) => {
-                                    errors.append(e);
-                                    continue;
-                                }
-                            };
 
-                            // These are definitely ValidatedCalls, but ValidatedCall::new()
-                            // returns ValidatedStatements, so they can be inserted directly
-                            for c in validated_calls {
-                                ret.insert(c);
+        for f in funcs.values() {
+            if f.is_associated_call {
+                let resource_name = match f.class {
+                    FunctionClass::Type(n) => n.name.clone(),
+                    _ => {
+                        // Can't derive from Global or API
```

Comment:
Would this be something user is trying to do erroneously that we should warn?

Reply:
As with below, the change here is just indent, so the comment is old.

I don't think it's possible to erroneously try to derive from a global function.  You have to inherit the parent to derive the function.  In terms of collections (not API anymore), I'd imagine there is/should be checks elsewhere.  Seems outside the scope of this PR though.  I'd think we'd warn on those back when we're parsing annotations on functions.